File: SICryption.py
from browser import window
import hashlib
import base64

class SICryption:
    def __init__(self):
        self.allowed_hash_alg = "sha512"
        self.active_sessions = {}

    # =========================
    # Encrypt / Decrypt Messages
    # =========================
    def hash_message(self, msg):
        return hashlib.sha512(msg.encode("utf-8")).hexdigest()

    def encode_base64(self, msg):
        return base64.b64encode(msg.encode("utf-8")).decode("utf-8")

    def decode_base64(self, msg):
        return base64.b64decode(msg.encode("utf-8")).decode("utf-8")

    # =========================
    # Session Validation
    # =========================
    def create_session(self, user_id):
        token = self.hash_message(user_id + "SECURE" + str(window.Date().now()))
        self.active_sessions[user_id] = token
        return token

    def validate_session(self, user_id, token):
        return self.active_sessions.get(user_id) == token

    # =========================
    # WatcherDog Integration
    # =========================
    def watcherDog(self, msg):
        mode = window.localStorage.getItem("mobby_mode") or "adult"
        blocked_words = ["hack", "kill", "password", "root", "<script>"]
        if mode == "kid":
            for word in blocked_words:
                if word in msg.lower():
                    return "WatcherDog Alert: Unsafe content blocked for Kid Mode."
        return msg

sicryption = SICryption()
window.sicryption = sicryption

File: SICryption.js
// ==========================
// SICryption.js – Client Guard
// ==========================

(() => {
  const ALLOWED_BROWSERS = ["chrome", "firefox", "brave", "safari", "edg"];
  const ALLOWED_HOST = "sic-corporation.github.io";

  const ua = navigator.userAgent.toLowerCase();
  const safeBrowser = ALLOWED_BROWSERS.some(b => ua.includes(b));

  if (!safeBrowser) {
    alert("🚫 Unsupported browser blocked by SICryption.");
    document.body.innerHTML = "<h1 style='color:red;text-align:center;margin-top:20%'>ACCESS DENIED</h1>";
    throw new Error("Browser blocked");
  }

  if (location.hostname !== ALLOWED_HOST && location.hostname !== "localhost") {
    document.body.innerHTML = "🚨 Unauthorized fork detected.";
    throw new Error("Fork blocked");
  }

  console.log("🔐 SICryption JS active");
})();

// ==========================
// JS Threat Scan
// ==========================
function jsThreatScan(message) {
  const patterns = [
    /<script/i,
    /eval\s*\(/i,
    /document\.cookie/i,
    /fetch\s*\(/i,
    /window\.location/i
  ];
  return !patterns.some(p => p.test(message));
}

// ==========================
// WatcherDog JS
// ==========================
function watcherDogJS(message) {
  const mode = localStorage.getItem("mobby_mode") || "adult";

  if (!jsThreatScan(message)) {
    alert("⚠️ WatcherDog blocked suspicious input.");
    return false;
  }

  if (mode === "kid" && /hack|kill|password|root/i.test(message)) {
    alert("🧸 Kid Mode Safety: message blocked.");
    return false;
  }

  return true;
}

window.watcherDogJS = watcherDogJS;
